cli/src/description_util.rs

```diff
@@ -31,12 +43,138 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
+            commit_builder.set_description(settings.default_description());
+        }
+        let temp_commit = commit_builder.write_hidden()?;
+        let template = description_template(tx, "", &temp_commit)?;
+        output_chunks.push(template);
+        output_chunks.push("\n".to_owned());
+    }
+    output_chunks
+        .push("JJ: Lines starting with \"JJ: \" (like this one) will be removed.\n".to_owned());
+    let bulk_message = output_chunks.join("");
+
+    let bulk_message = edit_temp_file(
+        "description",
+        ".jjdescription",
+        repo.repo_path(),
+        &bulk_message,
+        settings,
+    )?;
+
+    if commits.len() == 1 {
+        return Ok(EditMultipleDescriptionsResult {
+            descriptions: HashMap::from([(
+                commits[0].id().clone(),
+                cleanup_description(&bulk_message),
+            )]),
+            missing: vec![],
+            duplicates: vec![],
+            unexpected: vec![],
+        });
+    }
+
+    Ok(parse_bulk_edit_message(&bulk_message, &commits_map))
+}
+
+/// Parse the bulk message of edited commit descriptions.
+fn parse_bulk_edit_message(
+    message: &str,
+    commit_ids_map: &HashMap<String, &CommitId>,
+) -> EditMultipleDescriptionsResult {
+    let mut descriptions = HashMap::new();
+    let mut duplicates = Vec::new();
+    let mut unexpected = Vec::new();
+
+    let messages = message.lines().fold(vec![], |mut accum, line| {
+        if let Some(commit_id_prefix) = line.strip_prefix("JJ: describe ") {
+            let commit_id_prefix = commit_id_prefix
+                .strip_suffix(" -------")
+                .unwrap_or(commit_id_prefix);
+            accum.push((commit_id_prefix, vec![]));
+        } else if let Some((_, lines)) = accum.last_mut() {
+            lines.push(line);
+        };
+        accum
+    });
+
+    for (commit_id_prefix, description_lines) in messages {
+        let commit_id = match commit_ids_map.get(commit_id_prefix) {
+            Some(&commit_id) => commit_id,
+            None => {
+                unexpected.push(commit_id_prefix.to_string());
+                continue;
+            }
+        };
+        if descriptions.contains_key(commit_id) {
+            duplicates.push(commit_id_prefix.to_string());
+            continue;
+        }
+        descriptions.insert(
+            commit_id.clone(),
+            cleanup_description(&description_lines.join("\n")),
+        );
+    }
+
+    let missing: Vec<_> = commit_ids_map
+        .keys()
+        .filter_map(|commit_id_prefix| {
+            let commit_id = match commit_ids_map.get(commit_id_prefix) {
+                Some(&commit_id) => commit_id,
+                None => {
+                    return None;
+                }
+            };
+            if !descriptions.contains_key(commit_id) {
+                Some(commit_id_prefix.to_string())
+            } else {
+                None
+            }
+        })
+        .collect();
```

Comment:
nit: `missing` has to be sorted to stabilize the error message?

Might be better to process contents in the input `commits` order because the edited messages are usually sorted in that order.